examples/wasm-plugins/example-anchor-watch-dotnet/README.md

```diff
@@ -0,0 +1,715 @@
+
+[UnmanagedCallersOnly(EntryPoint = "plugin_name")]
+public static IntPtr GetName() { ... }
+
+[UnmanagedCallersOnly(EntryPoint = "plugin_schema")]
+public static IntPtr GetSchema() { ... }
+
+[UnmanagedCallersOnly(EntryPoint = "plugin_start")]
+public static int Start(IntPtr configPtr, int configLen) { ... }
+
+[UnmanagedCallersOnly(EntryPoint = "plugin_stop")]
+public static int Stop() { ... }
+```
+
+### FFI Bridge
+
+The plugin communicates with the Signal K server through FFI (Foreign Function Interface):
+
+```csharp
+[DllImport("env", EntryPoint = "sk_debug")]
+public static extern void Debug(IntPtr messagePtr, int messageLen);
+
+[DllImport("env", EntryPoint = "sk_register_put_handler")]
+public static extern int RegisterPutHandler(IntPtr contextPtr, int contextLen, IntPtr pathPtr, int pathLen);
+```
+
+### PUT Handler Registration
+
+During `plugin_start()`, the plugin registers its PUT handlers:
+
+```csharp
+SignalKApi.RegisterPut("vessels.self", "navigation.anchor.position");
+SignalKApi.RegisterPut("vessels.self", "navigation.anchor.maxRadius");
+SignalKApi.RegisterPut("vessels.self", "navigation.anchor.alarmState");
+```
+
+### PUT Handler Implementation
+
+Each PUT handler is exported with a specific naming convention:
+
+```csharp
+[UnmanagedCallersOnly(EntryPoint = "handle_put_vessels_self_navigation_anchor_position")]
+public static IntPtr HandleSetAnchorPosition(IntPtr requestPtr, int requestLen)
+{
+    // 1. Parse request JSON
+    var request = JsonSerializer.Deserialize<PutRequest>(requestJson);
+
+    // 2. Validate and process
+    var position = JsonSerializer.Deserialize<Position>(request.Value.GetRawText());
+    anchorState.Position = position;
+
+    // 3. Emit delta to update data model
+    SignalKApi.EmitDelta(deltaJson);
+
+    // 4. Return response
+    return MarshalJson(new PutResponse {
+        State = "COMPLETED",
+        StatusCode = 200,
+        Message = "Success"
+    });
+}
+```
+
+**Handler Naming Convention:**
+
+- Format: `handle_put_{context}_{path}` with dots replaced by underscores
+- Example: `handle_put_vessels_self_navigation_anchor_position`
+
+### Data Models
+
+The plugin uses strongly-typed C# classes with JSON serialization:
+
+```csharp
+public class Position
+{
+    [JsonPropertyName("latitude")]
+    public double Latitude { get; set; }
+
+    [JsonPropertyName("longitude")]
+    public double Longitude { get; set; }
+}
+
+public class PutRequest
+{
+    [JsonPropertyName("context")]
+    public string Context { get; set; }
+
+    [JsonPropertyName("path")]
+    public string Path { get; set; }
+
+    [JsonPropertyName("value")]
+    public JsonElement Value { get; set; }
+}
+
+public class PutResponse
+{
+    [JsonPropertyName("state")]
+    public string State { get; set; } = "COMPLETED";
+
+    [JsonPropertyName("statusCode")]
+    public int StatusCode { get; set; } = 200;
+
+    [JsonPropertyName("message")]
+    public string? Message { get; set; }
+}
+```
+
+## Configuration
```

Comment:
I don't think it is good practise to duplicate the configuration file format, the configuration parameter details nor invite end users to edit the files directly. The parameters are documented in the schema, and the more we duplicate the more we risk the real implementation and related documentation drifting apart over the lifetime of a component/plugin.

This goes for this .md as well as others.